### ISSUE TITLE
Final Shutdown on Google Play Music

### ISSUE THREAD

**Issue by MLisDreaming** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3903]
**Checklist:**

* [1] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES AS WELL**
* [1] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [1] I have double checked and can reproduce the issue

**OS:** Every OS

**GPMDP Version:** Every Version

**Issue Descriptions:** Google Play Music has official shutdown completely for music streaming.

**Steps to Reproduce:** Attempt to use GPM. We had a good run.

**Comment by gazsoimi** [comment]
https://techcrunch.com/2020/08/05/google-play-music-to-shut-down-starting-in-september-will-disappear-by-december/?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAGPtDJT4g6dPzDxtv9oGstYVZ0EsIaQUEn9rawpF9IWb7KsvfHRAnln6jKuyoRnXXKQt0zJjE4KW1QBKCepMaGRX479bo2FVvTH4-pZCf6l2MFBrJkXnuv8XONSU77-9Lku0znZuuwQYxCIlSMbjSlyRDueGXDQgi-M-LteJoGvh

**Comment by MLisDreaming** [comment]
IK

**Comment by andrewjmetzger** [comment]
See #3877, #3841, #3878. 

Hopefully, the maintainers of this project will drop GPM from the branding, and rework it into a YTM client. Until then, there is [YTM Desktop](https://github.com/ytmdesktop/ytmdesktop/tree/development), where the development branch (linked) has been working to prtovide some of the features GPMDP offored for users of Google Play Music. YTM Desktop is based on electron, and is slower, but until GPMDP is resurrected for YouTube Music, it seems like the least worst option.

**Comment by Dragon911XX** [comment]
I honestly would like to see the team work with iBroadcast (since it seems like that's where a lot of people have gone), as they don't currently have a desktop client setup. I absolutely loved GPMDP, and with them having the experience, I feel they could make something really good.

**Comment by andrewjmetzger** [comment]
Thanks for the mention of iBroadcast; it's always great to find out about new services. Out of curiosity, what drew you to iBrfoadcast vs. somewhere else? I've mostly been using YouTube music for discovery, along with [Plexamp](https://plexamp.com/), since I meant to store my music as FLACs anyway, and the current pandemic has given me ample time to re-organize my collection.

If you're comfortable with the command line, you could try turning `media.ibroadcast.com` into an Electron app with [Nativefier](https://github.com/jiahaog/Nativefier), which packages websites to give them a discrete window, desktop icon, and other features typically unique to native apps. Usually, it's as simple as `nativefier example.com`, although [the README](https://github.com/jiahaog/nativefier/blob/master/README.md) is always a good place to start. :smiley:

**Comment by Dragon911XX** [comment]
> Thanks for the mention of iBroadcast; it's always great to find out about new services. Out of curiosity, what drew you to iBrfoadcast vs. somewhere else? I've mostly been using YouTube music for discovery, along with [Plexamp](https://plexamp.com/), since I meant to store my music as FLACs anyway, and the current pandemic has given me ample time to re-organize my collection.
> 
> If you're comfortable with the command line, you could try turning `media.ibroadcast.com` into an Electron app with [Nativefier](https://github.com/jiahaog/Nativefier), which packages websites to give them a discrete window, desktop icon, and other features typically unique to native apps. Usually, it's as simple as `nativefier example.com`, although [the README](https://github.com/jiahaog/nativefier/blob/master/README.md) is always a good place to start. 😃

I first heard about iBroadcast actually on the Google Play Music subreddit. It was one of the things people found that worked really well. It drew me in because it's currently free (likely won't be after leaving the early stages). It's also nice that it allows me to upload all of my music. Something Spotify wont let me do. And it's such a massive pain to go through Spotify to add the music I already own to playlists, and end up listening to ads. Spotify is also really messy compared to iBroadcast. I really like the tag feature and how organized playlists and tags are. Not to mention the ability to sort my music.

Unfortunately, while GPM worked using my work's VPN, iBroadcast doesn't. Spotify does however. I've found a very small selection of music through Spotify, but most of what I listen to is Monstercat, VideoGame music, or things I find through Youtubers.

I actually did see Nativefier when searching for a way to use iBroadcast as a program. Since I'm mainly going to be listening to music in VR, where it seems Spotify has the best support for so far. Media Keys do work with iBroadcast, but it throws me off a bit not being able to see the album art (only the chrome icon), after seeing it so often with GPMDP. I'm interested to see how Nativefier works for this and in VR, but it seems too complex for me to set up.

**Comment by andrewjmetzger** [comment]
Nice music taste! I'm not so sure about VR -- never had it myself. If that's your use case, sticking with a big-name service is probably your easiest bet. If you have a Plex server, they do music too; [Plex VR](https://www.plex.tv/your-media/virtual-reality/) may do what you want. 

If you don't have a Plex server, throw it on your VR PC (presumably you have one). Plex isn't super resource-heavy, and it handles music like a dream. My library has the full Monstercat 1-30 collection and a bunch of OC Remix and lesser-known tracks; it works great. Lots of tags and organization, all in a nice folder structure straight from MusicBrainz and/or embedded metadata. You can, of  course, arrange it all however you'd like.

You could also try something like [Soundscape VR, available on Steam](https://store.steampowered.com/app/636930/Soundscape_VR/), which lets you play arbitrary music in VR... looks neat.

**Comment by Dragon911XX** [comment]
Plex was another one I looked at. I'm not sure how the Plex server works, if I need to pay a subscription. But the fact they have lots of other media options is cool. Although I'd only ever really use the Music stuff.

I just still need to redo all of my playlists over again which is tedious.

**Comment by MLisDreaming** [comment]
As much as it's nice to chat about the death of GPM, this issue is to discuss GPMDP's path after this. Please take these discussions to the GPM subreddit.


> See #3877, #3841, #3878.
> 
> Hopefully, the maintainers of this project will drop GPM from the branding, and rework it into a YTM client. Until then, there is [YTM Desktop](https://github.com/ytmdesktop/ytmdesktop/tree/development), where the development branch (linked) has been working to prtovide some of the features GPMDP offored for users of Google Play Music. YTM Desktop is based on electron, and is slower, but until GPMDP is resurrected for YouTube Music, it seems like the least worst option.

The two most likely courses of action seem to be to archive this entire repository or to work on the YTM features of this client.
We either can kill GPMDP or turn it into a purely YTMDP app. This player does have a base already with YTM being available in the current client.

**Comment by gabooh** [comment]
Shut down is now effective. YTM support works great. I hope this app continues to be developed.

**Comment by Delper** [comment]
> This player does have a base already with YTM being available in the current client.

The YTM player does not allow access to the desktop player settings.  All of the configuration options (output device, etc.) are unavailable.

**Comment by MLisDreaming** [comment]
> -If you did not switch over to the YTM interface before the shutdown was complete there is no way to do so now.
Incorrect, you can still switch using the icon in the notification tray.

**Comment by FinchPowers** [comment]
>>    -If you did not switch over to the YTM interface before the shutdown was complete there is no way to do so now.

> Incorrect, you can still switch using the icon in the notification tray.

In case readers are looking for it like I were: On MacOS this is the icon in the top menu bar near the clock. Not the one in the dock.

**Comment by Razzeee** [comment]
Doesn't seem to work with the flathub btw https://github.com/flathub/com.googleplaymusicdesktopplayer.GPMDP/issues/14
